riscv/src/continuations/memory_merkle_tree.rs

```diff
@@ -1,25 +1,46 @@
-use std::collections::BTreeMap;
+use std::{
+    any::TypeId,
+    collections::{BTreeMap, HashMap},
+    sync::{OnceLock, RwLock},
+};
 
 use super::bootloader::{
-    BYTES_PER_WORD, N_LEAVES_LOG as N_LEAVES_LOG_BOOTLOADER,
-    WORDS_PER_PAGE as WORDS_PER_PAGE_BOOTLOADER,
+    BYTES_PER_WORD, N_LEAVES_LOG, WORDS_PER_PAGE as WORDS_PER_PAGE_BOOTLOADER,
 };
+
 use powdr_number::FieldElement;
 use powdr_riscv_executor::poseidon_gl::poseidon_gl;
 
+const N_LEVELS_DEFAULT: usize = N_LEAVES_LOG + 1;
+
 /// A Merkle tree of memory pages.
 pub struct MerkleTree<
     T: FieldElement,
-    const N_LEAVES_LOG: usize = N_LEAVES_LOG_BOOTLOADER,
+    const N_LEVELS: usize = N_LEVELS_DEFAULT,
```

Comment:
I had to change this from `N_LEAVES_LOG` to `N_LEVELS` because the type of `default_hashes` can be `&'static [[T; 4]; N_LEVELS]`, but it can't be `&'static [[T; 4]; N_LEAVES_LOG + 1]`. Go figure...